src/core/scene/a-scene.js

```diff
@@ -709,7 +709,7 @@ function getMaxSize (maxSize, isVR) {
   var size;
   var pixelRatio = window.devicePixelRatio;
 
-  size = {height: window.innerHeight, width: window.innerWidth};
+  size = {height: document.body.offsetHeight, width: document.body.offsetWidth};
```

Comment:
Can you describe what's the difference with respect to `window.innerHeight`?

Reply:
That's a fair question, but I can't give a crystal clear answer. According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CSS_Object_Model/Determining_the_dimensions_of_elements), the `offsetWidth` and `offsetHeight` is the size of an element _including_ padding and border, but _excluing_ any margin.

The `<body>` element is always 100% wide AFAIK, but not necessarily 100% height. The height of a `<body>` is determined by what's inside it and how it's styled, so while `document.body.offsetWidth` should generally always be equal to `window.innerWidth`, `document.body.offsetHeight` is certainly not guaranteed to always be equal to `window.innerHeight`.

However, in this case we only use the `offsetWidth` and `offsetHeight` when the A-Frame scene is not embedded, meaning it should be styled to take up 100% width and height. This in turn should mean that the body has also been styled to a 100% width and height.

Finally, while is debatable whether `window.innerHeight` or `document.body.offsetHeight` is the best way to determine the height, it _does_ fix this very annoying bug in Mobile Safari, so I guess the question is if anyone has a non-standard setup where making this change will break their experience.